store/localcachelayer/user_layer.go

```diff
@@ -155,3 +157,62 @@ func (s LocalCacheUserStore) Get(id string) (*model.User, error) {
 	s.rootStore.doStandardAddToCache(s.rootStore.userProfileByIdsCache, id, user)
 	return user, nil
 }
+
+// GetMany is a cache wrapper around the SqlStore method to get a user profiles by ids.
+// It checks if the user entries are present in the cache, returning the entries from cache
+// if it is present. Otherwise, it fetches the entries from the store and stores it in the
+// cache.
+func (s LocalCacheUserStore) GetMany(ids []string) ([]*model.User, error) {
+	// we are doing a loop instead of caching the full set in the cache because the number of permutations that we can have
+	// in this func is making caching of the total set not beneficial.
+	var cachedUsers []*model.User
+	var notCachedUserIds []string
+	uniqIDs := dedup(ids)
```

Comment:
I think we can avoid the "double loop" here reusing the loop in the for, and storing the "already seen ids" in a map, that way you don't need a new function, and you don't need loop over the ids twice.

Reply:
Sure thing, I find the existing implementation easier to read IMO, and also they have the same asymptotic complexity as the one you are suggesting. 2/5 here since we do not take a perf hit and the code is more readable.

Reply:
Yes, my concern wasn't about performance in this case, I don't think is going to have any performance implications there, I was thinking more about readability, for me is easier to read the whole code in one single place, but in this case the "dedup" function is pretty self-explanatory, so... I not 100% sure about that. So... however yo prefer, it is ok for me. :)

Reply:
IMO existing implementation looks more readable.